third_party/terraform/data_sources/data_source_google_kms_crypto_key_version.go

```diff
@@ -0,0 +1,160 @@
+package google
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceGoogleKmsCryptoKeyVersion() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceGoogleKmsCryptoKeyVersionRead,
+		Schema: map[string]*schema.Schema{
+			"crypto_key": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"version": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  1,
+			},
+			"algorithm": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"protection_level": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"state": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"public_key": {
```

Comment:
As this is returned by a different method, and only present for asymmetric keys, we may want to break this out in its own data source, like `google_kms_crypto_key_version_public_key`.

Reply:
I could see an argument either way! On one hand, splitting it to a separate datasource means that information some users might consider sensitive won't be exposed. On the other, this feels like a nicer logical resource and I'd imagine the key is what users will want to retrieve most of the time.

I'll leave that choice to you.

Reply:
The main reason for breaking it into a different data source is avoiding the case in which the `google_kms_crypto_key_version_public_key` would not have a `public_key` (for example, if it's a symmetric KMS key). I don't know what's the general practice in terms of data sources with "optional" output fields. Any objections to that? I tried to make it clear in the data source documentation about when `public_key` would be present. The other reason to go with the explicit data source is to avoid the `public_key[0]` notation, which looks a bit silly to me. But from a user simplicity it make sense to keep a single data source, I'd say.

Reply:
The case where it's a bit silly to write `public_key[0]` isn't a big concern, given that we've gone so long needing to do that.

Fields that are only output conditionally are fine, we've done the same in several other places.

Reply:
Will keep it as it is, then. Thanks for the comments.